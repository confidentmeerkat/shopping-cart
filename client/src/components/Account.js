import React from 'react';
import moment from 'moment';
import numeral from 'numeral';
import { connect } from 'react-redux';
import { getUser } from '../actions/userActions';
import AccountModal from './AccountModal';
import Divider from 'material-ui/Divider';
import RaisedButton from 'material-ui/RaisedButton';
import Edit from 'material-ui/svg-icons/image/edit.js';
import '../styles/Account.css';

class Account extends React.Component {

  state = {
    accountModalOpen: false
  }

  toggleAccountModal = () => {
    this.setState({ accountModalOpen: !this.state.accountModalOpen });
  }

  componentDidMount() {
    this.props.getUser();
  }

  render() {
    if(!this.props.user) {
      return (
        <div className="account-container">
          <div className="loader">
            <img src="/img/loader.gif" />
            <h1>LOADING ACCOUNT DATA...</h1>
          </div>
        </div>
      );
    } else {
      return (
        <div className="account-container">
          <h1>Your Account</h1>
          <div className="account">
            <div className="account-info">
              <h2>Info</h2>
              <Divider />
              <p><b>Username: </b>{this.props.user.username}</p>
              <p><b>E-mail: </b>{this.props.user.email}</p>
              <p><b>Billing Address: </b>{this.props.user.address}</p>
              <p><b>Phone: </b>{this.props.user.phone}</p>
              <RaisedButton
                className="btn"
                label="Edit account"
                labelPosition="before"
                primary={true}
                icon={<Edit />}
                onClick={this.toggleAccountModal}
              />
            </div>
            <div className="account-history">
              <h2>Order History</h2>
              <Divider />
              <div class="orders">
                {this.props.user.orders.length ? 
                  <table>
                    <tr>
                      <th>Date Created</th>
                      <th>Product Name</th>
                      <th>Price</th>
                      <th>Qty</th>
                      <th>Total</th>
                    </tr>
                    {this.props.user.orders.map((order) => {
                      return (
                        <tr>
                          <td>{moment(order.dateCreated).format('ll')}</td>
                          <td>{order.name}</td>
                          <td>{numeral(order.price).format('$0,0.00')}</td>
                          <td>{order.quantity}</td>
                          <td>{numeral(order.price * order.quantity).format('$0,0.00')}</td>
                        </tr>
                      );
                    })}
                  </table> :
                  <h1>No order history.</h1>
                }
              </div>
            </div>
          </div>
          <AccountModal
            user={this.props.user}
            isOpen={this.state.accountModalOpen}
            onRequestClose={this.toggleAccountModal}
          />
        </div>
      );
    }
  }
};

const mapStateToProps = (state) => ({
  user: state.loggedUser
});

const mapDispatchToProps = (dispatch) => ({
  getUser: () => dispatch(getUser())
});

export default connect(mapStateToProps, mapDispatchToProps)(Account);