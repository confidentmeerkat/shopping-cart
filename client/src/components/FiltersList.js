import React from 'react';
import { connect } from 'react-redux';
import { List, ListItem } from 'material-ui/List';
import Subheader from 'material-ui/Subheader';
import Checkbox from 'material-ui/Checkbox';
import { setFilter } from '../actions/filterActions';
import '../styles/FiltersList.css';

const styles = {
  subheader: {
    fontSize: '16px',
    fontWeight: 'bold' 
  },
  listItem: {
    borderTop: '1px solid #dcdcdc'
  },
  checkbox: {
    marginLeft: 20
  }
}

class FiltersList extends React.Component {

  handleCheck = (e) => {
    this.props.setFilter(e.target.name, e.target.value);
  };

  render() {
    return (
      <div className="filtersList">
        <List>
          <Subheader style={styles.subheader}>Search by:</Subheader>
          <ListItem 
            style={styles.listItem}
            primaryText="Price"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox style={styles.checkbox} label="< $250" name="price" value="<250" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="$250 - $5000" name="price" value="250-500" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="$500 - $750" name="price" value="500-750" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="$750 >" name="price" value="750>" onCheck={this.handleCheck} />
            ]}
          />
          <ListItem 
            style={styles.listItem}
            primaryText="Brand"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox style={styles.checkbox} label="Samsung" name="brand" value="samsung" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="Apple" name="brand" value="apple" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="Huawei" name="brand" value="huawei" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="LG" name="brand" value="lg" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="HTC" name="brand" value="htc" onCheck={this.handleCheck} />,
            ]}
          />
          <ListItem 
            style={styles.listItem}
            primaryText="Color"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox style={styles.checkbox} label="Black" name="color" value="black" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="White" name="color" value="white" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="Grey" name="color" value="grey" onCheck={this.handleCheck} />
            ]}
          />
          <ListItem 
            style={styles.listItem}
            primaryText="OS"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox style={styles.checkbox} label="Android" name="os" value="android" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="iOS" name="os" value="ios" onCheck={this.handleCheck} />
            ]}
          />
          <ListItem 
            style={styles.listItem}
            primaryText="Internal memory"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox style={styles.checkbox} label="8GB" name="internalMemory" value="8" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="16GB" name="internalMemory" value="16" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="32GB" name="internalMemory" value="32" onCheck={this.handleCheck} />
            ]}
          />
          <ListItem 
            style={styles.listItem}
            primaryText="RAM"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox style={styles.checkbox} label="1GB" name="ram" value="1" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="2GB" name="ram" value="2" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="3GB" name="ram" value="3" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="4GB" name="ram" value="4" onCheck={this.handleCheck} />
            ]}
          />
          <ListItem 
            style={styles.listItem}
            primaryText="Display size"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox style={styles.checkbox} label="5''" name="displaySize" value="5" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="5.2''" name="displaySize" value="5.2" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="5.5''" name="displaySize" value="5.5" onCheck={this.handleCheck} />
            ]}
          />
          <ListItem 
            style={styles.listItem}
            primaryText="Display resolution"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox style={styles.checkbox} label="1280x720" name="displayResolution" value="1280x720" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="1920x1080" name="displayResolution" value="1920x1080" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="2560x1440" name="displayResolution" value="2560x1440" onCheck={this.handleCheck} />
            ]}
          />
          <ListItem 
            style={styles.listItem}
            primaryText="Camera"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox style={styles.checkbox} label="8Mpix" name="camera" value="8" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="12Mpix" name="camera" value="12" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="13Mpix" name="camera" value="13" onCheck={this.handleCheck} />
            ]}
          />
          <ListItem 
            style={styles.listItem}
            primaryText="CPU"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox style={styles.checkbox} label="Quad Core" name="cpu" value="quad core" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="Hexa Core" name="cpu" value="hexa core" onCheck={this.handleCheck} />,
              <Checkbox style={styles.checkbox} label="Octa Core" name="cpu" value="octa core" onCheck={this.handleCheck} />
            ]}
          />
        </List>
      </div>
    )
  }
};

const mapDispatchToProps = (dispatch) => ({
  setFilter: (filterType, filter) => dispatch(setFilter(filterType, filter))
});

export default connect(null, mapDispatchToProps)(FiltersList);