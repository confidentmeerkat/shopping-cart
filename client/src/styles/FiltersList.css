.filtersList {
  width: 20%;
  height:100%;
  min-height: auto;
  margin-top: 90px;
  margin-bottom: 40px;
  background-color: #f5f5f5;
}