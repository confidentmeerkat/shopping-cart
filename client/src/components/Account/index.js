import { compose } from 'recompose';
import { connect } from 'react-redux';
import { getUser } from '@actions';
import Account from './Account';

const mapStateToProps = (state) => ({
  user: state.loggedUser.user
});

const actions = { getUser };

export default compose(
  connect(mapStateToProps, actions)
)(Account);
