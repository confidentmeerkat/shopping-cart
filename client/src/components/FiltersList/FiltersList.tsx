import * as React from 'react';
import { List, ListItem } from 'material-ui/List';
import Subheader from 'material-ui/Subheader';
import Checkbox from 'material-ui/Checkbox';
import { IFilters } from '@state/index';
import '@styles/FiltersList.css';

interface Props {
  filters: IFilters;
  setFilter: (name: string, value: string) => void;
}

export class FiltersList extends React.Component<Props> {
  handleCheck = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { setFilter } = this.props;

    setFilter(e.target.name, e.currentTarget.value);
  };

  render() {
    const { filters: { checked } } = this.props;

    return (
      <div className="filtersList">
        <List>
          <Subheader className="subheader">Search by:</Subheader>
          <ListItem
            className="listItem"
            primaryText="Price"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox className="checkbox" label="< $250" name="priceRange" value="<250" checked={checked.includes('<250')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="$250 - $500" name="priceRange" value="250-500" checked={checked.includes('250-500')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="$500 - $750" name="priceRange" value="500-750" checked={checked.includes('500-750')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="$750 >" name="priceRange" value="750>" checked={checked.includes('750>')} onCheck={this.handleCheck} />
            ]}
          />
          <ListItem
            className="listItem"
            primaryText="Brand"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox className="checkbox" label="Samsung" name="brand" value="samsung" checked={checked.includes('samsung')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="Apple" name="brand" value="apple" checked={checked.includes('apple')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="Huawei" name="brand" value="huawei" checked={checked.includes('huawei')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="LG" name="brand" value="lg" checked={checked.includes('lg')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="HTC" name="brand" value="htc" checked={checked.includes('htc')} onCheck={this.handleCheck} />,
            ]}
          />
          <ListItem
            className="listItem"
            primaryText="Color"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox className="checkbox" label="Black" name="color" value="black" checked={checked.includes('black')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="White" name="color" value="white" checked={checked.includes('white')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="Grey" name="color" value="grey" checked={checked.includes('grey')} onCheck={this.handleCheck} />
            ]}
          />
          <ListItem
            className="listItem"
            primaryText="OS"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox className="checkbox" label="Android" name="os" value="android" checked={checked.includes('android')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="iOS" name="os" value="ios" checked={checked.includes('ios')} onCheck={this.handleCheck} />
            ]}
          />
          <ListItem
            className="listItem"
            primaryText="Internal memory"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox className="checkbox" label="16GB" name="internalMemory" value="16" checked={checked.includes('16')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="64GB" name="internalMemory" value="64" checked={checked.includes('64')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="128GB" name="internalMemory" value="128" checked={checked.includes('128')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="256GB" name="internalMemory" value="256" checked={checked.includes('256')} onCheck={this.handleCheck} />
            ]}
          />
          <ListItem
            className="listItem"
            primaryText="RAM"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox className="checkbox" label="1GB" name="ram" value="1" checked={checked.includes('1')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="3GB" name="ram" value="3" checked={checked.includes('3')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="4GB" name="ram" value="4" checked={checked.includes('4')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="6GB" name="ram" value="6" checked={checked.includes('6')} onCheck={this.handleCheck} />
            ]}
          />
          <ListItem
            className="listItem"
            primaryText="Display size"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox className="checkbox" label="4.5''" name="displaySize" value="4.5" checked={checked.includes('4.5')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="5.1''" name="displaySize" value="5.1" checked={checked.includes('5.1')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="5.5''" name="displaySize" value="5.5" checked={checked.includes('5.5')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="5.8''" name="displaySize" value="5.8" checked={checked.includes('5.8')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="6.0''" name="displaySize" value="6.0" checked={checked.includes('6.0')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="6.3''" name="displaySize" value="6.3" checked={checked.includes('6.3')} onCheck={this.handleCheck} />
            ]}
          />
          <ListItem
            className="listItem"
            primaryText="Display resolution"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox className="checkbox" label="540x960" name="displayResolution" value="540x960" checked={checked.includes('540x960')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="1080x1920" name="displayResolution" value="1080x1920" checked={checked.includes('1080x1920')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="1125x2436" name="displayResolution" value="1125x2436" checked={checked.includes('1125x2436')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="1440x2560" name="displayResolution" value="1440x2560" checked={checked.includes('1440x2560')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="1440x2880" name="displayResolution" value="1440x2880" checked={checked.includes('1440x2880')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="1440x2960" name="displayResolution" value="1440x2960" checked={checked.includes('1440x2960')} onCheck={this.handleCheck} />
            ]}
          />
          <ListItem
            className="listItem"
            primaryText="Camera"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox className="checkbox" label="8Mpix" name="camera" value="8" checked={checked.includes('8')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="12Mpix" name="camera" value="12" checked={checked.includes('12')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="13Mpix" name="camera" value="13" checked={checked.includes('13')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="16Mpix" name="camera" value="16" checked={checked.includes('16')} onCheck={this.handleCheck} />
            ]}
          />
          <ListItem
            className="listItem"
            primaryText="CPU"
            initiallyOpen={false}
            primaryTogglesNestedList={true}
            nestedItems={[
              <Checkbox className="checkbox" label="Quad Core" name="cpu" value="quad_core" checked={checked.includes('quad_core')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="Hexa Core" name="cpu" value="hexa_core" checked={checked.includes('hexa_core')} onCheck={this.handleCheck} />,
              <Checkbox className="checkbox" label="Octa Core" name="cpu" value="octa_core" checked={checked.includes('octa_core')} onCheck={this.handleCheck} />
            ]}
          />
        </List>
      </div>
    )
  }
};

export default FiltersList;
