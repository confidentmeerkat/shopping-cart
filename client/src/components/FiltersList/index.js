import { compose } from 'recompose';
import { connect } from 'react-redux';
import { selectFilters } from '@selectors/catalog';
import { setFilter } from '@actions';
import FiltersList from './FiltersList';

const mapStateToProps = (state) => ({
  filters: selectFilters(state)
});

const actions = { setFilter };

export default compose(
  connect(mapStateToProps, actions)
)(FiltersList);
