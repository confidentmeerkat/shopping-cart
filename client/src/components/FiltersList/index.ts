import { compose } from 'recompose';
import { connect } from 'react-redux';
import { selectFilters } from '@selectors/catalog';
import { setFilter } from '@actions/index';
import { IState } from '@typings/state/index';
import FiltersList from './FiltersList';

const mapStateToProps = (state: IState) => ({
  filters: selectFilters(state)
});

const actions = { setFilter };

export default compose(
  connect(mapStateToProps, actions)
)(FiltersList);
