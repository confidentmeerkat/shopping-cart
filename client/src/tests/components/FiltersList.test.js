import React from 'react';
import { shallow } from 'enzyme';
import { FiltersList } from '../../components/FiltersList';

describe('<FiltersList />', () => {
  it('renders FiltersList correctly', () => {
    const wrapper = shallow(<FiltersList />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.listItem')).toHaveLength(10);
  });
});