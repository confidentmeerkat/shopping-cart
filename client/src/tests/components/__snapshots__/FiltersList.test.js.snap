// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FiltersList /> renders FiltersList correctly 1`] = `
<div
  className="filtersList"
>
  <List>
    <Subheader
      className="subheader"
      inset={false}
    >
      Search by:
    </Subheader>
    <ListItem
      autoGenerateNestedIndicator={true}
      className="listItem"
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      nestedItems={
        Array [
          <Checkbox
            className="checkbox"
            disabled={false}
            label="< $250"
            labelPosition="right"
            name="priceRange"
            onCheck={[Function]}
            value="<250"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="$250 - $500"
            labelPosition="right"
            name="priceRange"
            onCheck={[Function]}
            value="250-500"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="$500 - $750"
            labelPosition="right"
            name="priceRange"
            onCheck={[Function]}
            value="500-750"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="$750 >"
            labelPosition="right"
            name="priceRange"
            onCheck={[Function]}
            value="750>"
          />,
        ]
      }
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Price"
      primaryTogglesNestedList={true}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      className="listItem"
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      nestedItems={
        Array [
          <Checkbox
            className="checkbox"
            disabled={false}
            label="Samsung"
            labelPosition="right"
            name="brand"
            onCheck={[Function]}
            value="samsung"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="Apple"
            labelPosition="right"
            name="brand"
            onCheck={[Function]}
            value="apple"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="Huawei"
            labelPosition="right"
            name="brand"
            onCheck={[Function]}
            value="huawei"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="LG"
            labelPosition="right"
            name="brand"
            onCheck={[Function]}
            value="lg"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="HTC"
            labelPosition="right"
            name="brand"
            onCheck={[Function]}
            value="htc"
          />,
        ]
      }
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Brand"
      primaryTogglesNestedList={true}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      className="listItem"
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      nestedItems={
        Array [
          <Checkbox
            className="checkbox"
            disabled={false}
            label="Black"
            labelPosition="right"
            name="color"
            onCheck={[Function]}
            value="black"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="White"
            labelPosition="right"
            name="color"
            onCheck={[Function]}
            value="white"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="Grey"
            labelPosition="right"
            name="color"
            onCheck={[Function]}
            value="grey"
          />,
        ]
      }
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Color"
      primaryTogglesNestedList={true}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      className="listItem"
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      nestedItems={
        Array [
          <Checkbox
            className="checkbox"
            disabled={false}
            label="Android"
            labelPosition="right"
            name="os"
            onCheck={[Function]}
            value="android"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="iOS"
            labelPosition="right"
            name="os"
            onCheck={[Function]}
            value="ios"
          />,
        ]
      }
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="OS"
      primaryTogglesNestedList={true}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      className="listItem"
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      nestedItems={
        Array [
          <Checkbox
            className="checkbox"
            disabled={false}
            label="16GB"
            labelPosition="right"
            name="internalMemory"
            onCheck={[Function]}
            value="16"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="64GB"
            labelPosition="right"
            name="internalMemory"
            onCheck={[Function]}
            value="64"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="128GB"
            labelPosition="right"
            name="internalMemory"
            onCheck={[Function]}
            value="128"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="256GB"
            labelPosition="right"
            name="internalMemory"
            onCheck={[Function]}
            value="256"
          />,
        ]
      }
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Internal memory"
      primaryTogglesNestedList={true}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      className="listItem"
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      nestedItems={
        Array [
          <Checkbox
            className="checkbox"
            disabled={false}
            label="1GB"
            labelPosition="right"
            name="ram"
            onCheck={[Function]}
            value="1"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="3GB"
            labelPosition="right"
            name="ram"
            onCheck={[Function]}
            value="3"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="4GB"
            labelPosition="right"
            name="ram"
            onCheck={[Function]}
            value="4"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="6GB"
            labelPosition="right"
            name="ram"
            onCheck={[Function]}
            value="6"
          />,
        ]
      }
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="RAM"
      primaryTogglesNestedList={true}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      className="listItem"
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      nestedItems={
        Array [
          <Checkbox
            className="checkbox"
            disabled={false}
            label="4.5''"
            labelPosition="right"
            name="displaySize"
            onCheck={[Function]}
            value="4.5"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="5.1''"
            labelPosition="right"
            name="displaySize"
            onCheck={[Function]}
            value="5.1"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="5.5''"
            labelPosition="right"
            name="displaySize"
            onCheck={[Function]}
            value="5.5"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="5.8''"
            labelPosition="right"
            name="displaySize"
            onCheck={[Function]}
            value="5.8"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="6.0''"
            labelPosition="right"
            name="displaySize"
            onCheck={[Function]}
            value="6.0"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="6.3''"
            labelPosition="right"
            name="displaySize"
            onCheck={[Function]}
            value="6.3"
          />,
        ]
      }
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Display size"
      primaryTogglesNestedList={true}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      className="listItem"
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      nestedItems={
        Array [
          <Checkbox
            className="checkbox"
            disabled={false}
            label="540x960"
            labelPosition="right"
            name="displayResolution"
            onCheck={[Function]}
            value="540x960"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="1080x1920"
            labelPosition="right"
            name="displayResolution"
            onCheck={[Function]}
            value="1080x1920"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="1125x2436"
            labelPosition="right"
            name="displayResolution"
            onCheck={[Function]}
            value="1125x2436"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="1440x2560"
            labelPosition="right"
            name="displayResolution"
            onCheck={[Function]}
            value="1440x2560"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="1440x2880"
            labelPosition="right"
            name="displayResolution"
            onCheck={[Function]}
            value="1440x2880"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="1440x2960"
            labelPosition="right"
            name="displayResolution"
            onCheck={[Function]}
            value="1440x2960"
          />,
        ]
      }
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Display resolution"
      primaryTogglesNestedList={true}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      className="listItem"
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      nestedItems={
        Array [
          <Checkbox
            className="checkbox"
            disabled={false}
            label="8Mpix"
            labelPosition="right"
            name="camera"
            onCheck={[Function]}
            value="8"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="12Mpix"
            labelPosition="right"
            name="camera"
            onCheck={[Function]}
            value="12"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="13Mpix"
            labelPosition="right"
            name="camera"
            onCheck={[Function]}
            value="13"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="16Mpix"
            labelPosition="right"
            name="camera"
            onCheck={[Function]}
            value="16"
          />,
        ]
      }
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Camera"
      primaryTogglesNestedList={true}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      className="listItem"
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      nestedItems={
        Array [
          <Checkbox
            className="checkbox"
            disabled={false}
            label="Quad Core"
            labelPosition="right"
            name="cpu"
            onCheck={[Function]}
            value="quad_core"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="Hexa Core"
            labelPosition="right"
            name="cpu"
            onCheck={[Function]}
            value="hexa_core"
          />,
          <Checkbox
            className="checkbox"
            disabled={false}
            label="Octa Core"
            labelPosition="right"
            name="cpu"
            onCheck={[Function]}
            value="octa_core"
          />,
        ]
      }
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="CPU"
      primaryTogglesNestedList={true}
      secondaryTextLines={1}
    />
  </List>
</div>
`;
